script/test_in_openshift/lib.sh

```diff
@@ -0,0 +1,231 @@
+	convert::print_fail "oc cluster up failed.\n"
+	exit $exit_status
+    fi
+
+    convert::run_cmd "oc login -u system:admin"
+}
+
+function convert::oc_cluster_down () {
+
+    convert::run_cmd "oc cluster down"
+    exit_status=$?
+
+    if [ $exit_status -ne 0 ]; then
+	FAIL_MSGS=$FAIL_MSGS"exit status: $exit_status\n"
+	exit $exit_status
+    fi
+
+}
+
+function convert::kompose_up () {
+    dc_file=$1
+    kompose_cli='kompose --emptyvols --provider=openshift -f $dc_file up'
+    convert::run_cmd "${kompose_cli}"
+    exit_status=$?
+
+    if [ $exit_status -ne 0 ]; then
+	FAIL_MSGS=$FAIL_MSGS"kompose up failed with exit status: $exit_status\n"
+	exit $exit_status
+    fi
+}
+
+function convert::kompose_down () {
+    dc_file=$1
+    kompose_cli='kompose --emptyvols --provider=openshift -f $dc_file down'
+    convert::run_cmd $kompose_cli
+    exit_status=$?
+
+    if [ $exit_status -ne 0 ]; then
+	FAIL_MSGS=$FAIL_MSGS"kompose down failed with exit status: $exit_status\n";
+	exit $exit_status;
+    fi
+}
+
+
+function convert::kompose_up_check () {
+    # Usage: -p for pod name, -r replica count
+    local retry_up=0
+   
+    while getopts ":p:r:" opt; do
+	case $opt in
+	    p ) pod=$OPTARG;;
+	    r ) replica=$OPTARG;;
+	esac
+    done
+
+    if [ -z $replica ]; then
+       replica_1=1
+       replica_2=1
+    else
+       replica_1=$replica
+       replica_2=$replica
+    fi
+    
+    pod_1=$( echo $pod | awk '{ print $1 }')
+    pod_2=$( echo $pod | awk '{ print $2 }')
+    
+    query_1='grep ${pod_1} | grep -v deploy'
+    query_2='grep ${pod_2} | grep -v deploy'
+    
+    query_1_status='Running'
+    query_2_status='Running'
+    
+    is_buildconfig=$(oc get builds --no-headers | wc -l)
+
+    if [ $is_buildconfig -gt 0 ]; then
+	query_1='grep ${pod_1} | grep -v deploy | grep -v build'
+	query_2='grep build | grep -v deploy'
+	query_2_status='Completed'
+	replica_2=1 
+    fi
+
+    convert::print_msg "Waiting for the pods to come up ..."
+    
+    # FIXME: Make this generic to cover all cases
+    while [ $(oc get pods | eval ${query_1} | awk '{ print $3 }' | \
+		     grep ${query_1_status} | wc -l) -ne $replica_1 ] ||
+	      [ $(oc get pods | eval ${query_2} | awk '{ print $3 }' | \
+			 grep ${query_2_status} | wc -l) -ne $replica_2 ]; do
+	
+	if [ $retry_up -lt 240 ]; then
+	    retry_up=$(($retry_up + 1))
+	    sleep 1
+	else
+	    convert::print_fail "kompose up has failed to bring the pods up\n"
+	    oc get pods
+	    exit 1
+	fi
+	
+    done
+
+    # Wait
+    sleep 2
+
+    # If pods are up, print a success message
+    if [ $(oc get pods | eval ${query_1} | awk '{ print $3 }' | \
+		  grep ${query_1_status} | wc -l) -eq $replica_1 ] &&
+	   [ $(oc get pods | eval ${query_2} | awk '{ print $3 }' | \
+		      grep ${query_2_status} | wc -l) -eq $replica_2 ]; then
+	oc get pods
+	convert::print_pass "All pods are Running now. kompose up is successful."
+    fi
+}
+
+function convert::kompose_down_check () {
```

Comment:
can you please add comment explaining what parameter is required for this function and what it means?